Naming update for Stacks rebrand #841

friedger opened the pull request:
## Description

This PR replaces the use of the name `Blockstack` with `Stacks` where it makes sense.

Some naming changes require an additional noun like `Blockstack` in the context of authentication becomes `Stacks Auth`.

For details refer to issue #841 

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [x] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
No

## Are documentation updates required?
No

## Testing information
Nothing special

## Checklist
- [x] Code is commented where needed
- [x] Unit test coverage for new or modified code paths
- [x] `npm run test` passes
- [x] Changelog is updated
- [x] Tag 1 of @agraebe @GinaAbrams

GinaAbrams commented on src/pages/data-storage/collections.md:
```diff
@@ -18,7 +18,7 @@ particular. The following topics are covered:
 ## Understand how collections work
 
-One of Blockstack's goals is to give users true data ownership by enabling _data portability_. Data portability allows
+One of Stacks's goals is to give users true data ownership by enabling _data portability_. Data portability allows
```
One of the goals of the Stacks ecosystem is to give

GinaAbrams commented on src/pages/data-storage/collections.md:
```diff
@@ -43,15 +43,15 @@ pre-release provides the `Contact` collection. A contact schema produces this st
 
-Because malicious apps or apps with poor security controls may damage user data, Blockstack believes collections should
-include the ability for users to roll-back changes. For this reason, Blockstack supports an event log and rollback
+Because malicious apps or apps with poor security controls may damage user data, Stacks believes collections should
```
This seems funky, can we remove mentions of "Stacks believes"

GinaAbrams commented on src/pages/data-storage/overview.md:
```diff
@@ -1,39 +1,39 @@
 Blockchains require consensus among large numbers of people, so they can be slow. Additionally, a blockchain is not designed to hold a lot of data. This means using a blockchain for every bit of data a user might write and store is expensive. For example, imagine if an application were storing every tweet in the chain.
 
-Blockstack addresses blockchain performance problems using a layered approach. The base layer consists of the Stacks blockchain and the Blockstack Naming System (BNS). The blockchain governs ownership of identities in the Blockstack network. Identities can be names such as domain names, usernames, or application names.
+Stacks addresses blockchain performance problems using a layered approach. The base layer consists of the Stacks blockchain and the Blockstack Naming System (BNS). The blockchain governs ownership of identities in the Stacks network. Identities can be names such as domain names, usernames, or application names.
```
The Stacks blockchain addresses

GinaAbrams commented on src/pages/data-storage/overview.md:
```diff
@@ -1,39 +1,39 @@
 then writes to storage on behalf of that user.
 
-Within Blockstack, then, the Stacks blockchain stores only identity data. Data created by the actions of an identity is stored in a Gaia Storage System. Each user has profile data. When a user interacts with a decentralized dApp that application stores application data on behalf of the user. Because Gaia stores user and application data off the blockchain, a Blockstack DApp is typically more perf ...
+Within Stacks, then, the Stacks blockchain stores only identity data. Data created by the actions of an identity is stored in a Gaia Storage System. Each user has profile data. When a user interacts with a decentralized dApp that application stores application data on behalf of the user. Because Gaia stores user and application data off the blockchain, a Stacks DApp is typically more performant t ...
```
The Stacks blockchain stores only

GinaAbrams commented on src/pages/authentication/overview.md:
```diff
@@ -120,18 +120,18 @@ The app private key is securely shared with the app on each authentication, encr
 Both the `authRequest` and the `authResponse` tokens are [JSON Web Tokens](https://jwt.io/), and they are passed via URL query strings.
 
-Blockstack's authentication tokens are based on the [RFC 7519 OAuth JSON Web Token (JWT)](https://tools.ietf.org/html/rfc7519)
+Stacks's authentication tokens are based on the [RFC 7519 OAuth JSON Web Token (JWT)](https://tools.ietf.org/html/rfc7519)
```
Stacks authentication tokens?

GinaAbrams commented:
Added a couple small comments, looks good otherwise! 🙏

friedger commented:
@GinaAbrams there is a paragraph about Blockstack PBC and the history. What should be done with it?

Also who issued the stacks tokens?

friedger commented:
@GinaAbrams  I have addressed your comments.

What about "The Blockstack App" and the "Blockstack Browser"?

The the two section in the overview:

* Blockstack PBC section: https://github.com/blockstack/docs/blob/05ac3f2882773346cfc1a9f371132219d3ce1810/src/pages/ecosystem/overview.md#L16
* Blockstack Token LLC: https://github.com/blockstack/docs/blob/05ac3f2882773346cfc1a9f371132219d3ce1810/src/pages/ecosystem/overview.md#L38

GinaAbrams commented:
Thanks Friedger! I think we can probably wait on making changes to the PBC overview or Token LLC overview until the Hiro change has been more finalized.

agraebe commented:
@GinaAbrams could you help with a final review? I think we could merge this today if all looks good

GinaAbrams commented on src/pages/data-storage/collections.md:
```diff
@@ -18,13 +18,13 @@ particular. The following topics are covered:
 decentralized ID.
 
-How do collections work? Blockstack builds a library containing commonly used data schemes. Developers use these classes
+How do collections work? Hiro PBC builds a library containing commonly used data schemes. Developers use these classes
```
Can we use Stacks here to remain consistent on the language 🙏

friedger replied:
Updated!

GinaAbrams reviewed (commented):
Added a single small comment. Otherwise lgtm!

agraebe commented:
fantastic, thanks for this @friedger. I'll merge once the conflicts are resolved. can you take a look?

friedger commented:
done

agraebe commented:
@aulneau could you help here as well? I can't merge this one for some reason

agraebe commented:
@friedger I can't merge this PR because of a conflict. could you resolve/rebase so we can merge this? thanks!

friedger commented:
@agraebe done

CharlieC3 commented:
@friedger I think I see what's going on WRT the merge conflict. There's been some commits to `master` since you created this PR, in order to get a smooth merge and fix this, you need to perform a manual _rebase_ in your feature branch. Unfortunately it seems the `master` branch was merged into this feature branch quite a few times. There's a way to fix this properly, but honestly with the amount of merges from `master` performed it'll be simpler to just save the changes you made in this PR and re-create the feature branch from the latest commit in `master`.

For future reference, when you want to pull updates from the `master` branch into your feature branch, use:
`git pull --rebase origin master`

This will lift up the stack of changes you've made in the feature branch, and tuck new master updates underneath them to maintain chronological order, then it places your commits back on top of the commit stack.

You'll then need to _force_ push your branch back up to your fork
`git push <my-fork-name> -f`

I do this everytime I pull in updates from a base branch, and it should help keep PRs clean/mergeable.

friedger commented:
This PR is replaced by #900
